### PR TITLE
statistics: refactor TableCacheItem

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/45367

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by qw4990 — APPROVED** [review]

File: statistics/column.go [hunk 1]
```diff
@@ -352,38 +352,18 @@ func (c *Column) DropEvicted() {
...
-		// For stats version2, there is no cms thus we directly drop topn
-		c.dropTopN()
-		return
-	case onlyCmsEvicted:
-		c.dropTopN()
+		c.dropUnnecessaryData()
 		return
 	default:
 		return
 	}
```
Comment (winoros) [inline statistics/column.go:359]: Can it be simplified with the upon if branch?

File: statistics/index.go [hunk 2]
```diff
@@ -61,33 +61,21 @@ func (idx *Index) IsAllEvicted() bool {
...
-	idx.evictedStatus = onlyCmsEvicted
+func (idx *Index) getEvictedStatus() int {
+	return idx.evictedStatus
 }
 
-func (idx *Index) dropHist() {
+func (idx *Index) dropUnnecessaryData() {
+	if idx.isCMSExist() && idx.statsVer() < Version2 {
+		idx.CMSketch = nil
+	}
```
Comment (winoros) [inline statistics/index.go:71]: Same with the Column's

**Pull request merged**